ta4j-core/src/main/java/org/ta4j/core/TickBarBuilder.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    @Override
+    public BarBuilder lowPrice(final Num lowPrice) {
+        throw new IllegalArgumentException("Ticks are build just from closePrices");
+    }
+
+    @Override
+    public BarBuilder lowPrice(final Number lowPrice) {
+        throw new IllegalArgumentException("Ticks are build just from closePrices");
+    }
+
+    @Override
+    public BarBuilder lowPrice(final String lowPrice) {
+        throw new IllegalArgumentException("Ticks are build just from closePrices");
+    }
+
+    @Override
+    public BarBuilder closePrice(final Num tickPrice) {
+        this.closePrice = tickPrice;
+        if (this.openPrice == null) {
+            this.openPrice = tickPrice;
+        }
+
+        this.highPrice = this.highPrice.max(tickPrice);
+        this.lowPrice = this.lowPrice.min(tickPrice);
+
+        return this;
+    }
+
+    @Override
+    public BarBuilder closePrice(final Number closePrice) {
+        return closePrice(this.numFactory.numOf(closePrice));
+    }
+
+    @Override
+    public BarBuilder closePrice(final String closePrice) {
+        return closePrice(this.numFactory.numOf(closePrice));
+    }
+
+    @Override
+    public BarBuilder volume(final Num volume) {
+        this.volume = this.volume.plus(volume);
+        return this;
+    }
+
+    @Override
+    public BarBuilder volume(final Number volume) {
+        volume(this.numFactory.numOf(volume));
+        return this;
+    }
+
+    @Override
+    public BarBuilder volume(final String volume) {
+        volume(this.numFactory.numOf(volume));
+        return this;
+    }
+
+    @Override
+    public BarBuilder amount(final Num amount) {
+        this.amount = amount;
+        return this;
+    }
+
+    @Override
+    public BarBuilder amount(final Number amount) {
+        amount(this.numFactory.numOf(amount));
+        return this;
+    }
+
+    @Override
+    public BarBuilder amount(final String amount) {
+        amount(this.numFactory.numOf(amount));
+        return this;
+    }
+
+    @Override
+    public BarBuilder trades(final long trades) {
+        this.trades = trades;
+        return this;
+    }
+
+    @Override
+    public BarBuilder trades(final String trades) {
+        trades(Long.parseLong(trades));
+        return this;
+    }
+
+    @Override
+    public BarBuilder bindTo(final BarSeries barSeries) {
+        this.barSeries = Objects.requireNonNull(barSeries);
+        return this;
+    }
+
+    /**
+     * Builds bar from current state that is modified for each tick.
+     *
+     * @return snapshot of current state
+     */
+    @Override
+    public Bar build() {
+        return new BaseBar(this.timePeriod, this.endTime, this.openPrice, this.highPrice, this.lowPrice,
+                this.closePrice, this.volume, this.amount, this.trades);
+    }
+
+    @Override
+    public void add() {
+        if (++this.countOfTicksPassed % this.countOfTicks == 0) {
+            this.barSeries.addBar(build());
+            this.openPrice = null;
```

Comment:
This "reset" sequence could be in it's own method as it's done in multiple places. If it were 3+ places I would ask for it explicitly, but if it's only 2 I'll leave it to your discretion.

Help me understand the rationale of the differing reset values. `openPrice` and `closePrice` are reset to null, but `highPrice` and `lowPrice` are set to semantically "nonsense" values. Why the difference?

I don't do tick-based trading myself. What is the significance of checking `if (++this.countOfTicksPassed % this.countOfTicks == 0)` rather than a strict equality check. Wouldn't this allow bars with varying multiples of ticks?

Reply:
"nonsense" values are chosen for min/max algorithm. High goes from zero to actual high and low goes from "infinity" to actual value.

Open/close are set to null as undefined value and no need of calculation.

For equality check I would need to reset countOfTicksPassed to zero. Modulo is reset free.